alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt

```diff
@@ -219,39 +220,18 @@ object DocumentLevelMonitorRunner : MonitorRunner() {
                     // Prepare DocumentExecutionContext for each index
                     val docExecutionContext = DocumentExecutionContext(queries, indexLastRunContext, indexUpdatedRunContext)
 
-                    val matchingDocs = getMatchingDocs(
+                    fetchDataAndExecutePercolateQueriesPerShard(
                         monitor,
                         monitorCtx,
                         docExecutionContext,
                         updatedIndexName,
                         concreteIndexName,
                         conflictingFields.toList(),
-                        matchingDocIdsPerIndex?.get(concreteIndexName)
+                        matchingDocIdsPerIndex?.get(concreteIndexName),
+                        monitorMetadata,
+                        inputRunResults,
+                        docsToQueries
```

Comment:
we keep collecting doc ids to queries here & run 1 finding creation, trigger execution workflow https://github.com/opensearch-project/alerting/pull/1331/files#diff-68866b22ed9703814b4d5db8d3488872bcb972086ecaca10c9b8bfd54db981bcR237.
Do we also execute finding creation, trigger execution workflow at each shard level?

Reply:
trigger execution workflows needs all results.
Findings we can brainstorm if it needs to be done at shard level
But currently findings are being indexed one by one. A bigger and more important optimization would be to ingest them batch wise using bulk requests: https://github.com/opensearch-project/alerting/issues/1333